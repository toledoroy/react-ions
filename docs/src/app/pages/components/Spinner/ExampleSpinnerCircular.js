import React from 'react'
import Spinner from 'react-ions/lib/components/Spinner'

const ExampleSpinnerCircular = () => (
  <Spinner loading={true} type='spinner-circular' color='#3C97D3' size='24' />
)

export default ExampleSpinnerCircular
