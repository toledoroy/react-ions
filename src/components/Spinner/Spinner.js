import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import styles from './styles.css'

export const StyledSpan = styled.span`${props => props.css}`

class Spinner extends React.Component {
  constructor(props) {
    super(props)

    if (props.optClass && process.env.NODE_ENV !== 'production') {
      console.warn('Dropdown: Use of optClass will be deprecated as of react-ions 6.0.0, please use `className` instead')
    }
  }

  state = {
    loading: false
  }

  static propTypes = {
    /**
     * Delay before showing spinner (in milliseconds)
     */
    delay: PropTypes.number,
    /**
     * Whether the spinner is displayed
     */
    loading: PropTypes.bool,
    /**
     * CSS positioning options for the loader. By default, the spinner will be positioned
     * in the center of any element with relative positioning.
     */
    position: PropTypes.oneOf(['fixed', 'inline']),
    /**
     * The type of loader you want to display.
     */
    type: PropTypes.oneOf(['spinner-dots', 'spinner-bounce', 'spinner-circular']).isRequired,
    /**
     * The hex code of the background color.
     */
    backgroundColor: PropTypes.string,
    /**
     * The hex code of the loader color.
     */
    color: PropTypes.string,
    /**
     * Optional styles to add to the spinner.
     */
    optClass: PropTypes.string,
    /**
     * A class name to be used for local styles or integrations (required to support styled-components)
     */
    className: PropTypes.string,
    /**
     * The size of the spinner.
     */
    size: PropTypes.string
  }

  UNSAFE_componentWillMount = () => {
    this.getLoadingState(this.props)
  }

  UNSAFE_componentWillReceiveProps = nextProps => {
    this.getLoadingState(nextProps)
  }

  shouldComponentUpdate = (nextProps, nextState) => {
    return (
      nextState.loading !== this.state.loading
    )
  }

  getLoadingState = props => {
    if (props.loading && props.delay) {
      this.timeout = setTimeout(() => {
        this.setState({ loading: true })
      }, props.delay)
    } else {
      clearTimeout(this.timeout)
      this.setState({ loading: props.loading || false })
    }
  }

  innerHtml = () => {
    if (this.props.type === 'spinner-dots') {
      return (
        <span>
          <span className='dot1'></span>
          <span className='dot2'></span>
        </span>
      )
    }

    if (this.props.type === 'spinner-circular') {
      return (
        <div className='sc-inner'>
          <div className='sc-inner-animation'>
            <svg width={this.props.size || '20'} height={this.props.size || '20'} strokeWidth='16.00' viewBox='-3.00 -3.00 106.00 106.00'>
              <path
                className='sc-inner-track'
                d='M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90'>
              </path>
              <path
                className='sc-inner-head'
                d='M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90'
                pathLength='280'
                strokeDasharray='280 280'
                strokeDashoffset='210'>
              </path>
            </svg>
          </div>
        </div>
      )
    }

    return (
      <span>
        <span className='bounce1'></span>
        <span className='bounce2'></span>
        <span className='bounce3'></span>
      </span>
    )
  }

  render = () => {
    const { backgroundColor, className, color, optClass, position, size, type } = this.props;
    const { loading } = this.state;

    console.log("STYLE", styles({ loading, hidden: !loading, position: position || 'absolute', color, size, backgroundColor }));

    return (
      <StyledSpan
        css={styles({ loading, hidden: !loading, position: position || 'absolute', color, size, backgroundColor })}
        className={[optClass, className].join(' ').trim()}
      >
        <span className={type}>
          {this.innerHtml()}
        </span>
      </StyledSpan>
    )
  }
}

export default Spinner
