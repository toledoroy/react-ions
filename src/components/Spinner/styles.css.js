import { keyframes, css } from 'styled-components'
import colors from '../../styles/common/colors.css'
import { zLayers } from '../../styles/common/functions.css'

const dotsRotate = keyframes`
  100% {
    transform: rotate(360deg);
  }
`

const dotsBounce = keyframes`
  0%, 100% {
    transform: scale(0.0);
  } 50% {
    transform: scale(1.0);
  }
`

const bounceDelay = keyframes`
  0%, 80%, 100% {
    transform: scale(0);
  } 40% {
    transform: scale(1.0);
  }
`

const ring = keyframes`
  0% {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(1turn);
  }
`

const sharedSpinnerSettings = `
  align-items: center;
  bottom: 0;
  justify-content: center;
  left: 0;
  right: 0;
  top: 0;
`

const loadingStyles = loading => loading ? `display: flex;` : ``
const hiddenStyles = hidden => hidden ? `display: none;` : ``
const positionStyles = position => {
  if (position === 'inline') return `display: inline-flex;`
  if (position === 'fixed') {
    return `
      ${sharedSpinnerSettings}
      position: fixed;
      z-index: ${zLayers['spinner-fixed']};
    `
  }
  if (position === 'absolute') {
    return `
      ${ sharedSpinnerSettings }
      position: absolute;
      z-index: ${zLayers['spinner-absolute']};
    `
  }
}

const styles = ({ backgroundColor, color, hidden, loading, position, size }) => {
  return css`
    background-color: ${backgroundColor || colors.white};
    display: none;
    ${loadingStyles(loading)}
    ${positionStyles(position)}
    ${hiddenStyles(!loading)}

    .spinner-dots {
      animation: ${dotsRotate} 2.0s infinite linear;
      height: ${size || 40}px;
      position: relative;
      width: ${size || 40}px;
    }

    .spinner-dots .dot1, .dot2 {
      animation: ${dotsBounce} 2.0s infinite ease-in-out;
      border-radius: 100%;
      display: inline-block;
      height: 60%;
      position: absolute;
      top: 0;
      width: 60%;
    }

    .spinner-dots .dot1 {
      background-color: ${color || colors.primary_1};
      height: 60%;
      top: 0;
    }

    .spinner-dots .dot2 {
      animation-delay: -1.0s;
      background-color: ${color || colors.danger};
      bottom: 0;
      height: 60%;
      top: auto;
    }

    .spinner-bounce .bounce1, .bounce2, .bounce3 {
      animation: ${bounceDelay} 1.4s infinite ease-in-out both;
      background-color: ${color || colors.neutral_4};
      border-radius: 100%;
      display: inline-block;
      height: ${size || 12}px;
      margin-right: 3px;
      width: ${size || 12}px;
    }

    .spinner-bounce .bounce1 {
      animation-delay: -0.32s;
    }

    .spinner-bounce .bounce2 {
      animation-delay: -0.16s;
    }

    .spinner-bounce .bounce3 {
      margin-right: 0;
    }

    .spinner-circular {
      display: inline-block;
      position: relative;
      width: ${size || 16}px;
      height: ${size || 16}px;
    }

    .spinner-circular .sc-inner {
      display: flex;
      align-items: center;
      justify-content: center;
      overflow: visible;
      vertical-align: middle;
    }

    .spinner-circular .sc-inner-animation {
        animation: ${ring} .5s linear infinite;
    }

    .spinner-circular .sc-inner svg {
        display: block;
    }

    .spinner-circular .sc-inner path {
        fill-opacity: 0;
    }

    .spinner-circular .sc-inner-head {
        transform-origin: center;
        transition: stroke-dashoffset .2s cubic-bezier(.4, 1, .75, .9);
        stroke: ${color || colors.neutral_4};
        stroke-linecap: round;
    }

    .spinner-circular .sc-inner-track {
        stroke: rgba(92, 112, 128, .2);
    }
  `
}

export default styles