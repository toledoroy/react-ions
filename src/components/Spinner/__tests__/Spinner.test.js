import React from 'react'
import Spinner from '../Spinner'

describe('Spinner dots', () => {
  it('displays a spinner-dots class', () => {
    const spinner = shallow(<Spinner type='spinner-dots' />)
    expect(spinner).toMatchSnapshot()
  })
})

describe('Spinner bounce', () => {
  it('displays a spinner-bounce class', () => {
    const spinner = shallow(<Spinner type='spinner-bounce' />)
    expect(spinner).toMatchSnapshot()
  })
})

describe('Spinner circular', () => {
  it('displays a spinner-circular class', () => {
    const spinner = shallow(<Spinner type='spinner-circular' />)
    expect(spinner).toMatchSnapshot()
  })
})

describe('Spinner with loading true', () => {
  it('has a loading class', () => {
    const spinner = shallow(<Spinner loading={true} type='spinner-bounce' />)
    expect(spinner).toMatchSnapshot()
  })
})

describe('Spinner with fixed position', () => {
  it('has a loading class', () => {
    const spinner = shallow(<Spinner loading={true} type='spinner-bounce' position='fixed' />)
    expect(spinner).toMatchSnapshot()
  })
})

describe('Spinner with inline position', () => {
  it('has a loading class', () => {
    const spinner = shallow(<Spinner loading={true} type='spinner-bounce' position='inline' />)
    expect(spinner).toMatchSnapshot()
  })
})

describe('Spinner', () => {
  it('has an className', () => {
    const spinner = shallow(<Spinner loading={true} type='spinner-bounce' className='testing' />)
    expect(spinner).toMatchSnapshot()
  })
})

describe('Spinner', () => {
  it('has custom background color', () => {
    const spinner = shallow(<Spinner loading={true} type='spinner-bounce' color='#3C97D3' />)
    expect(spinner).toMatchSnapshot()
  })
})

describe('Spinner', () => {
  it('displays an is-hidden class', () => {
    const spinner = shallow(<Spinner loading={false} type='spinner-bounce' />)
    expect(spinner).toMatchSnapshot()
  })
})

describe('Spinner', () => {
  jest.useFakeTimers()

  it('calls a setTimeout when delay is passed', () => {
    const spinner = shallow(<Spinner loading={true} delay={1000} type='spinner-bounce' />)

    expect(setTimeout).toHaveBeenCalledTimes(1)
    expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 1000)
  })
})
