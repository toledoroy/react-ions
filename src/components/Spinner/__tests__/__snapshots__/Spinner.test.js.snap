// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner bounce displays a spinner-bounce class 1`] = `
<styled.span
  className=""
  css={
    Object {
      ".spinner-bounce": Object {
        ".bounce1": Object {
          "animationDelay": "-0.32s",
        },
        ".bounce1, .bounce2, .bounce3": Object {
          "animation": "ghjfcd 1.4s infinite ease-in-out both",
          "backgroundColor": "#9198A0",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "12px",
          "marginRight": "3px",
          "width": "12px",
        },
        ".bounce2": Object {
          "animationDelay": "-0.16s",
        },
        ".bounce3": Object {
          "marginRight": 0,
        },
      },
      ".spinner-circular": Object {
        ".sc-inner": Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "overflow": "visible",
          "verticalAlign": "middle",
        },
        ".sc-inner path": Object {
          "fillOpacity": "0",
        },
        ".sc-inner svg": Object {
          "display": "block",
        },
        ".sc-inner-animation": Object {
          "animation": "gWPqvk .5s linear infinite",
        },
        ".sc-inner-head": Object {
          "stroke": "#9198A0",
          "strokeLinecap": "round",
          "transformOrigin": "center",
          "transition": "stroke-dashoffset .2s cubic-bezier(.4, 1, .75, .9)",
        },
        ".sc-inner-track": Object {
          "stroke": "rgba(92, 112, 128, .2)",
        },
        "display": "inline-block",
        "height": "16px",
        "position": "relative",
        "width": "16px",
      },
      ".spinner-dots": Object {
        ".dot1, .dot2": Object {
          "animation": "iQfAUB 2.0s infinite ease-in-out",
          "backgroundColor": "#3C97D3",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "60%",
          "position": "absolute",
          "top": 0,
          "width": "60%",
        },
        ".dot2": Object {
          "animationDelay": "-1.0s",
          "backgroundColor": "#E54C3B",
          "bottom": 0,
          "top": "auto",
        },
        "animation": "cZxgpV 2.0s infinite linear",
        "height": "40px",
        "position": "relative",
        "width": "40px",
      },
      "alignItems": "center",
      "backgroundColor": "#fff",
      "bottom": 0,
      "display": "none",
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <span
    className="spinner-bounce"
  >
    <span>
      <span
        className="bounce1"
      />
      <span
        className="bounce2"
      />
      <span
        className="bounce3"
      />
    </span>
  </span>
</styled.span>
`;

exports[`Spinner circular displays a spinner-circular class 1`] = `
<styled.span
  className=""
  css={
    Object {
      ".spinner-bounce": Object {
        ".bounce1": Object {
          "animationDelay": "-0.32s",
        },
        ".bounce1, .bounce2, .bounce3": Object {
          "animation": "ghjfcd 1.4s infinite ease-in-out both",
          "backgroundColor": "#9198A0",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "12px",
          "marginRight": "3px",
          "width": "12px",
        },
        ".bounce2": Object {
          "animationDelay": "-0.16s",
        },
        ".bounce3": Object {
          "marginRight": 0,
        },
      },
      ".spinner-circular": Object {
        ".sc-inner": Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "overflow": "visible",
          "verticalAlign": "middle",
        },
        ".sc-inner path": Object {
          "fillOpacity": "0",
        },
        ".sc-inner svg": Object {
          "display": "block",
        },
        ".sc-inner-animation": Object {
          "animation": "gWPqvk .5s linear infinite",
        },
        ".sc-inner-head": Object {
          "stroke": "#9198A0",
          "strokeLinecap": "round",
          "transformOrigin": "center",
          "transition": "stroke-dashoffset .2s cubic-bezier(.4, 1, .75, .9)",
        },
        ".sc-inner-track": Object {
          "stroke": "rgba(92, 112, 128, .2)",
        },
        "display": "inline-block",
        "height": "16px",
        "position": "relative",
        "width": "16px",
      },
      ".spinner-dots": Object {
        ".dot1, .dot2": Object {
          "animation": "iQfAUB 2.0s infinite ease-in-out",
          "backgroundColor": "#3C97D3",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "60%",
          "position": "absolute",
          "top": 0,
          "width": "60%",
        },
        ".dot2": Object {
          "animationDelay": "-1.0s",
          "backgroundColor": "#E54C3B",
          "bottom": 0,
          "top": "auto",
        },
        "animation": "cZxgpV 2.0s infinite linear",
        "height": "40px",
        "position": "relative",
        "width": "40px",
      },
      "alignItems": "center",
      "backgroundColor": "#fff",
      "bottom": 0,
      "display": "none",
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <span
    className="spinner-circular"
  >
    <div
      className="sc-inner"
    >
      <div
        className="sc-inner-animation"
      >
        <svg
          height="20"
          strokeWidth="16.00"
          viewBox="-3.00 -3.00 106.00 106.00"
          width="20"
        >
          <path
            className="sc-inner-track"
            d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
          />
          <path
            className="sc-inner-head"
            d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
            pathLength="280"
            strokeDasharray="280 280"
            strokeDashoffset="210"
          />
        </svg>
      </div>
    </div>
  </span>
</styled.span>
`;

exports[`Spinner displays an is-hidden class 1`] = `
<styled.span
  className=""
  css={
    Object {
      ".spinner-bounce": Object {
        ".bounce1": Object {
          "animationDelay": "-0.32s",
        },
        ".bounce1, .bounce2, .bounce3": Object {
          "animation": "ghjfcd 1.4s infinite ease-in-out both",
          "backgroundColor": "#9198A0",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "12px",
          "marginRight": "3px",
          "width": "12px",
        },
        ".bounce2": Object {
          "animationDelay": "-0.16s",
        },
        ".bounce3": Object {
          "marginRight": 0,
        },
      },
      ".spinner-circular": Object {
        ".sc-inner": Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "overflow": "visible",
          "verticalAlign": "middle",
        },
        ".sc-inner path": Object {
          "fillOpacity": "0",
        },
        ".sc-inner svg": Object {
          "display": "block",
        },
        ".sc-inner-animation": Object {
          "animation": "gWPqvk .5s linear infinite",
        },
        ".sc-inner-head": Object {
          "stroke": "#9198A0",
          "strokeLinecap": "round",
          "transformOrigin": "center",
          "transition": "stroke-dashoffset .2s cubic-bezier(.4, 1, .75, .9)",
        },
        ".sc-inner-track": Object {
          "stroke": "rgba(92, 112, 128, .2)",
        },
        "display": "inline-block",
        "height": "16px",
        "position": "relative",
        "width": "16px",
      },
      ".spinner-dots": Object {
        ".dot1, .dot2": Object {
          "animation": "iQfAUB 2.0s infinite ease-in-out",
          "backgroundColor": "#3C97D3",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "60%",
          "position": "absolute",
          "top": 0,
          "width": "60%",
        },
        ".dot2": Object {
          "animationDelay": "-1.0s",
          "backgroundColor": "#E54C3B",
          "bottom": 0,
          "top": "auto",
        },
        "animation": "cZxgpV 2.0s infinite linear",
        "height": "40px",
        "position": "relative",
        "width": "40px",
      },
      "alignItems": "center",
      "backgroundColor": "#fff",
      "bottom": 0,
      "display": "none",
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <span
    className="spinner-bounce"
  >
    <span>
      <span
        className="bounce1"
      />
      <span
        className="bounce2"
      />
      <span
        className="bounce3"
      />
    </span>
  </span>
</styled.span>
`;

exports[`Spinner dots displays a spinner-dots class 1`] = `
<styled.span
  className=""
  css={
    Object {
      ".spinner-bounce": Object {
        ".bounce1": Object {
          "animationDelay": "-0.32s",
        },
        ".bounce1, .bounce2, .bounce3": Object {
          "animation": "ghjfcd 1.4s infinite ease-in-out both",
          "backgroundColor": "#9198A0",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "12px",
          "marginRight": "3px",
          "width": "12px",
        },
        ".bounce2": Object {
          "animationDelay": "-0.16s",
        },
        ".bounce3": Object {
          "marginRight": 0,
        },
      },
      ".spinner-circular": Object {
        ".sc-inner": Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "overflow": "visible",
          "verticalAlign": "middle",
        },
        ".sc-inner path": Object {
          "fillOpacity": "0",
        },
        ".sc-inner svg": Object {
          "display": "block",
        },
        ".sc-inner-animation": Object {
          "animation": "gWPqvk .5s linear infinite",
        },
        ".sc-inner-head": Object {
          "stroke": "#9198A0",
          "strokeLinecap": "round",
          "transformOrigin": "center",
          "transition": "stroke-dashoffset .2s cubic-bezier(.4, 1, .75, .9)",
        },
        ".sc-inner-track": Object {
          "stroke": "rgba(92, 112, 128, .2)",
        },
        "display": "inline-block",
        "height": "16px",
        "position": "relative",
        "width": "16px",
      },
      ".spinner-dots": Object {
        ".dot1, .dot2": Object {
          "animation": "iQfAUB 2.0s infinite ease-in-out",
          "backgroundColor": "#3C97D3",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "60%",
          "position": "absolute",
          "top": 0,
          "width": "60%",
        },
        ".dot2": Object {
          "animationDelay": "-1.0s",
          "backgroundColor": "#E54C3B",
          "bottom": 0,
          "top": "auto",
        },
        "animation": "cZxgpV 2.0s infinite linear",
        "height": "40px",
        "position": "relative",
        "width": "40px",
      },
      "alignItems": "center",
      "backgroundColor": "#fff",
      "bottom": 0,
      "display": "none",
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <span
    className="spinner-dots"
  >
    <span>
      <span
        className="dot1"
      />
      <span
        className="dot2"
      />
    </span>
  </span>
</styled.span>
`;

exports[`Spinner has an className 1`] = `
<styled.span
  className="testing"
  css={
    Object {
      ".spinner-bounce": Object {
        ".bounce1": Object {
          "animationDelay": "-0.32s",
        },
        ".bounce1, .bounce2, .bounce3": Object {
          "animation": "ghjfcd 1.4s infinite ease-in-out both",
          "backgroundColor": "#9198A0",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "12px",
          "marginRight": "3px",
          "width": "12px",
        },
        ".bounce2": Object {
          "animationDelay": "-0.16s",
        },
        ".bounce3": Object {
          "marginRight": 0,
        },
      },
      ".spinner-circular": Object {
        ".sc-inner": Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "overflow": "visible",
          "verticalAlign": "middle",
        },
        ".sc-inner path": Object {
          "fillOpacity": "0",
        },
        ".sc-inner svg": Object {
          "display": "block",
        },
        ".sc-inner-animation": Object {
          "animation": "gWPqvk .5s linear infinite",
        },
        ".sc-inner-head": Object {
          "stroke": "#9198A0",
          "strokeLinecap": "round",
          "transformOrigin": "center",
          "transition": "stroke-dashoffset .2s cubic-bezier(.4, 1, .75, .9)",
        },
        ".sc-inner-track": Object {
          "stroke": "rgba(92, 112, 128, .2)",
        },
        "display": "inline-block",
        "height": "16px",
        "position": "relative",
        "width": "16px",
      },
      ".spinner-dots": Object {
        ".dot1, .dot2": Object {
          "animation": "iQfAUB 2.0s infinite ease-in-out",
          "backgroundColor": "#3C97D3",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "60%",
          "position": "absolute",
          "top": 0,
          "width": "60%",
        },
        ".dot2": Object {
          "animationDelay": "-1.0s",
          "backgroundColor": "#E54C3B",
          "bottom": 0,
          "top": "auto",
        },
        "animation": "cZxgpV 2.0s infinite linear",
        "height": "40px",
        "position": "relative",
        "width": "40px",
      },
      "alignItems": "center",
      "backgroundColor": "#fff",
      "bottom": 0,
      "display": "flex",
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <span
    className="spinner-bounce"
  >
    <span>
      <span
        className="bounce1"
      />
      <span
        className="bounce2"
      />
      <span
        className="bounce3"
      />
    </span>
  </span>
</styled.span>
`;

exports[`Spinner has custom background color 1`] = `
<styled.span
  className=""
  css={
    Object {
      ".spinner-bounce": Object {
        ".bounce1": Object {
          "animationDelay": "-0.32s",
        },
        ".bounce1, .bounce2, .bounce3": Object {
          "animation": "ghjfcd 1.4s infinite ease-in-out both",
          "backgroundColor": "#3C97D3",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "12px",
          "marginRight": "3px",
          "width": "12px",
        },
        ".bounce2": Object {
          "animationDelay": "-0.16s",
        },
        ".bounce3": Object {
          "marginRight": 0,
        },
      },
      ".spinner-circular": Object {
        ".sc-inner": Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "overflow": "visible",
          "verticalAlign": "middle",
        },
        ".sc-inner path": Object {
          "fillOpacity": "0",
        },
        ".sc-inner svg": Object {
          "display": "block",
        },
        ".sc-inner-animation": Object {
          "animation": "gWPqvk .5s linear infinite",
        },
        ".sc-inner-head": Object {
          "stroke": "#3C97D3",
          "strokeLinecap": "round",
          "transformOrigin": "center",
          "transition": "stroke-dashoffset .2s cubic-bezier(.4, 1, .75, .9)",
        },
        ".sc-inner-track": Object {
          "stroke": "rgba(92, 112, 128, .2)",
        },
        "display": "inline-block",
        "height": "16px",
        "position": "relative",
        "width": "16px",
      },
      ".spinner-dots": Object {
        ".dot1, .dot2": Object {
          "animation": "iQfAUB 2.0s infinite ease-in-out",
          "backgroundColor": "#3C97D3",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "60%",
          "position": "absolute",
          "top": 0,
          "width": "60%",
        },
        ".dot2": Object {
          "animationDelay": "-1.0s",
          "backgroundColor": "#3C97D3",
          "bottom": 0,
          "top": "auto",
        },
        "animation": "cZxgpV 2.0s infinite linear",
        "height": "40px",
        "position": "relative",
        "width": "40px",
      },
      "alignItems": "center",
      "backgroundColor": "#fff",
      "bottom": 0,
      "display": "flex",
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <span
    className="spinner-bounce"
  >
    <span>
      <span
        className="bounce1"
      />
      <span
        className="bounce2"
      />
      <span
        className="bounce3"
      />
    </span>
  </span>
</styled.span>
`;

exports[`Spinner with fixed position has a loading class 1`] = `
<styled.span
  className=""
  css={
    Object {
      ".spinner-bounce": Object {
        ".bounce1": Object {
          "animationDelay": "-0.32s",
        },
        ".bounce1, .bounce2, .bounce3": Object {
          "animation": "ghjfcd 1.4s infinite ease-in-out both",
          "backgroundColor": "#9198A0",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "12px",
          "marginRight": "3px",
          "width": "12px",
        },
        ".bounce2": Object {
          "animationDelay": "-0.16s",
        },
        ".bounce3": Object {
          "marginRight": 0,
        },
      },
      ".spinner-circular": Object {
        ".sc-inner": Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "overflow": "visible",
          "verticalAlign": "middle",
        },
        ".sc-inner path": Object {
          "fillOpacity": "0",
        },
        ".sc-inner svg": Object {
          "display": "block",
        },
        ".sc-inner-animation": Object {
          "animation": "gWPqvk .5s linear infinite",
        },
        ".sc-inner-head": Object {
          "stroke": "#9198A0",
          "strokeLinecap": "round",
          "transformOrigin": "center",
          "transition": "stroke-dashoffset .2s cubic-bezier(.4, 1, .75, .9)",
        },
        ".sc-inner-track": Object {
          "stroke": "rgba(92, 112, 128, .2)",
        },
        "display": "inline-block",
        "height": "16px",
        "position": "relative",
        "width": "16px",
      },
      ".spinner-dots": Object {
        ".dot1, .dot2": Object {
          "animation": "iQfAUB 2.0s infinite ease-in-out",
          "backgroundColor": "#3C97D3",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "60%",
          "position": "absolute",
          "top": 0,
          "width": "60%",
        },
        ".dot2": Object {
          "animationDelay": "-1.0s",
          "backgroundColor": "#E54C3B",
          "bottom": 0,
          "top": "auto",
        },
        "animation": "cZxgpV 2.0s infinite linear",
        "height": "40px",
        "position": "relative",
        "width": "40px",
      },
      "alignItems": "center",
      "backgroundColor": "#fff",
      "bottom": 0,
      "display": "flex",
      "justifyContent": "center",
      "left": 0,
      "position": "fixed",
      "right": 0,
      "top": 0,
      "zIndex": 4000,
    }
  }
>
  <span
    className="spinner-bounce"
  >
    <span>
      <span
        className="bounce1"
      />
      <span
        className="bounce2"
      />
      <span
        className="bounce3"
      />
    </span>
  </span>
</styled.span>
`;

exports[`Spinner with inline position has a loading class 1`] = `
<styled.span
  className=""
  css={
    Object {
      ".spinner-bounce": Object {
        ".bounce1": Object {
          "animationDelay": "-0.32s",
        },
        ".bounce1, .bounce2, .bounce3": Object {
          "animation": "ghjfcd 1.4s infinite ease-in-out both",
          "backgroundColor": "#9198A0",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "12px",
          "marginRight": "3px",
          "width": "12px",
        },
        ".bounce2": Object {
          "animationDelay": "-0.16s",
        },
        ".bounce3": Object {
          "marginRight": 0,
        },
      },
      ".spinner-circular": Object {
        ".sc-inner": Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "overflow": "visible",
          "verticalAlign": "middle",
        },
        ".sc-inner path": Object {
          "fillOpacity": "0",
        },
        ".sc-inner svg": Object {
          "display": "block",
        },
        ".sc-inner-animation": Object {
          "animation": "gWPqvk .5s linear infinite",
        },
        ".sc-inner-head": Object {
          "stroke": "#9198A0",
          "strokeLinecap": "round",
          "transformOrigin": "center",
          "transition": "stroke-dashoffset .2s cubic-bezier(.4, 1, .75, .9)",
        },
        ".sc-inner-track": Object {
          "stroke": "rgba(92, 112, 128, .2)",
        },
        "display": "inline-block",
        "height": "16px",
        "position": "relative",
        "width": "16px",
      },
      ".spinner-dots": Object {
        ".dot1, .dot2": Object {
          "animation": "iQfAUB 2.0s infinite ease-in-out",
          "backgroundColor": "#3C97D3",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "60%",
          "position": "absolute",
          "top": 0,
          "width": "60%",
        },
        ".dot2": Object {
          "animationDelay": "-1.0s",
          "backgroundColor": "#E54C3B",
          "bottom": 0,
          "top": "auto",
        },
        "animation": "cZxgpV 2.0s infinite linear",
        "height": "40px",
        "position": "relative",
        "width": "40px",
      },
      "backgroundColor": "#fff",
      "display": "inline-flex",
    }
  }
>
  <span
    className="spinner-bounce"
  >
    <span>
      <span
        className="bounce1"
      />
      <span
        className="bounce2"
      />
      <span
        className="bounce3"
      />
    </span>
  </span>
</styled.span>
`;

exports[`Spinner with loading true has a loading class 1`] = `
<styled.span
  className=""
  css={
    Object {
      ".spinner-bounce": Object {
        ".bounce1": Object {
          "animationDelay": "-0.32s",
        },
        ".bounce1, .bounce2, .bounce3": Object {
          "animation": "ghjfcd 1.4s infinite ease-in-out both",
          "backgroundColor": "#9198A0",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "12px",
          "marginRight": "3px",
          "width": "12px",
        },
        ".bounce2": Object {
          "animationDelay": "-0.16s",
        },
        ".bounce3": Object {
          "marginRight": 0,
        },
      },
      ".spinner-circular": Object {
        ".sc-inner": Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "overflow": "visible",
          "verticalAlign": "middle",
        },
        ".sc-inner path": Object {
          "fillOpacity": "0",
        },
        ".sc-inner svg": Object {
          "display": "block",
        },
        ".sc-inner-animation": Object {
          "animation": "gWPqvk .5s linear infinite",
        },
        ".sc-inner-head": Object {
          "stroke": "#9198A0",
          "strokeLinecap": "round",
          "transformOrigin": "center",
          "transition": "stroke-dashoffset .2s cubic-bezier(.4, 1, .75, .9)",
        },
        ".sc-inner-track": Object {
          "stroke": "rgba(92, 112, 128, .2)",
        },
        "display": "inline-block",
        "height": "16px",
        "position": "relative",
        "width": "16px",
      },
      ".spinner-dots": Object {
        ".dot1, .dot2": Object {
          "animation": "iQfAUB 2.0s infinite ease-in-out",
          "backgroundColor": "#3C97D3",
          "borderRadius": "100%",
          "display": "inline-block",
          "height": "60%",
          "position": "absolute",
          "top": 0,
          "width": "60%",
        },
        ".dot2": Object {
          "animationDelay": "-1.0s",
          "backgroundColor": "#E54C3B",
          "bottom": 0,
          "top": "auto",
        },
        "animation": "cZxgpV 2.0s infinite linear",
        "height": "40px",
        "position": "relative",
        "width": "40px",
      },
      "alignItems": "center",
      "backgroundColor": "#fff",
      "bottom": 0,
      "display": "flex",
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "zIndex": 1000,
    }
  }
>
  <span
    className="spinner-bounce"
  >
    <span>
      <span
        className="bounce1"
      />
      <span
        className="bounce2"
      />
      <span
        className="bounce3"
      />
    </span>
  </span>
</styled.span>
`;
